homeassistant/components/vizio/media_player.py

```diff
@@ -127,100 +140,130 @@ def update(self):
+    def source_list(self) -> List:
         """Return list of available inputs of the device."""
+
         return self._available_inputs
 
     @property
-    def supported_features(self):
+    def supported_features(self) -> int:
         """Flag device features that are supported."""
+
         return self._supported_commands
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return the unique id of the device."""
+
         return self._unique_id
 
-    def turn_on(self):
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device registry information."""
+
+        return {
+            "identifiers": {(DOMAIN, self._unique_id)},
+            "name": self.name,
+            "manufacturer": "Vizio",
+        }
+
+    async def async_turn_on(self) -> None:
         """Turn the device on."""
-        self._device.pow_on()
 
-    def turn_off(self):
+        await self._device.pow_on()
+
+    async def async_turn_off(self) -> None:
         """Turn the device off."""
-        self._device.pow_off()
 
-    def mute_volume(self, mute):
+        await self._device.pow_off()
+
+    async def async_mute_volume(self, mute: bool) -> None:
         """Mute the volume."""
+
         if mute:
-            self._device.mute_on()
+            await self._device.mute_on()
         else:
-            self._device.mute_off()
+            await self._device.mute_off()
 
-    def media_previous_track(self):
+    async def async_media_previous_track(self) -> None:
         """Send previous channel command."""
-        self._device.ch_down()
 
-    def media_next_track(self):
+        await self._device.ch_down()
+
+    async def async_media_next_track(self) -> None:
         """Send next channel command."""
-        self._device.ch_up()
 
-    def select_source(self, source):
+        await self._device.ch_up()
+
+    async def async_select_source(self, source: str) -> None:
         """Select input source."""
-        self._device.input_switch(source)
 
-    def volume_up(self):
+        await self._device.input_switch(source)
+
+    async def async_volume_up(self) -> None:
         """Increasing volume of the device."""
-        self._device.vol_up(num=self._volume_step)
+
+        await self._device.vol_up(self._volume_step)
+
         if self._volume_level is not None:
             self._volume_level = min(
                 1.0, self._volume_level + self._volume_step / self._max_volume
             )
 
-    def volume_down(self):
+    async def async_volume_down(self) -> None:
         """Decreasing volume of the device."""
-        self._device.vol_down(num=self._volume_step)
+
+        await self._device.vol_down(self._volume_step)
+
         if self._volume_level is not None:
             self._volume_level = max(
                 0.0, self._volume_level - self._volume_step / self._max_volume
             )
 
-    def validate_setup(self):
+    async def validate_setup(self) -> bool:
```

Comment:
In a future PR it would be good to remove this method and do the validation using the device before instantiating the entity. Then pass the device to the entity.